server/api.go

```diff
@@ -47,6 +53,66 @@ type SessionResult struct {
 	LegacySession bool `json:"-"` // true if request was started with legacy (i.e. pre-condiscon) session request
 }
 
+// SessionDisclosureResult contains an extended array of generalized credentials with the zkp
+type SessionDisclosureResult struct {
+	Identifier  *big.Int                            `json:"identifier"` // Identifier assigned by verifier
+	Token       irma.RequestorToken                 `json:"token"`
+	Requestor   SessionDisclosureResultRequestor    `json:"requestor"`
+	Nonce       *big.Int                            `json:"nonce"` // Actual nonce used in proofs
+	Status      irma.ProofStatus                    `json:"status"`
+	Timestamp   *atum.Timestamp                     `json:"timestamp,omitempty"` // If present, timestamp is included in nonce (see ASN1ConvertSignatureNonce)
+	Credentials []SessionDisclosureResultCredential `json:"credentials"`
+}
+
+// DisclosureResult input
+type SessionDisclosureResultRequestor struct {
+	Message *string  `json:"message,omitempty"` // If present, message is included in nonce (see ASN1ConvertSignatureNonce)
+	Nonce   *big.Int `json:"nonce,omitempty"`   // If present, this nonce is used as base nonce
```

Comment:
I moved the identifier to the requestor struct and refactored `Requestor` to `Request` instead of your suggestion. See also comment below on identifier.